contracts/instruments/RibbonThetaVault.sol

```diff
@@ -262,26 +283,52 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
      * @param share is the number of vault shares to be burned
      */
     function withdraw(uint256 share) external nonReentrant {
-        uint256 withdrawAmount = _withdraw(share);
+        uint256 withdrawAmount = _withdraw(share, false);
         IERC20(asset).safeTransfer(msg.sender, withdrawAmount);
     }
 
     /**
      * @notice Burns vault shares and checks if eligible for withdrawal
      * @param share is the number of vault shares to be burned
+     * @param isScheduled is whether the withdraw was scheduled
      */
-    function _withdraw(uint256 share) private returns (uint256) {
+    function _withdraw(uint256 share, bool isScheduled)
+        private
+        returns (uint256)
+    {
         (uint256 amountAfterFee, uint256 feeAmount) =
             withdrawAmountWithShares(share);
 
         emit Withdraw(msg.sender, amountAfterFee, share, feeAmount);
 
-        _burn(msg.sender, share);
+        _burn(isScheduled ? address(this) : msg.sender, share);
+
+        if (isYearnWrapped) {
+            _unwrapYieldToken(amountAfterFee.add(feeAmount));
+        }
+
         IERC20(asset).safeTransfer(feeRecipient, feeAmount);
 
         return amountAfterFee;
     }
 
+    /**
+     * @notice Unwraps the necessary amount of the yield-bearing yearn token
+     *         and transfers amount to relevant recipient
+     * @param amount is the amount of `asset` to withdraw
+     */
+    function _unwrapYieldToken(uint256 amount) private returns (uint256) {
+        uint256 assetBalance = IERC20(asset).balanceOf(address(this));
+        uint256 amountToUnwrap =
+            max(assetBalance, amount).sub(assetBalance).div(
+                collateralToken.pricePerShare()
+            );
```

Comment:
This could be 0 because of the division,

```
x = x < 10**18
amountToUnwrap = x / 10**18 = 0
```

Need to use `wdiv` for these cases